src/types.jl

```diff
@@ -235,317 +235,281 @@ mutable struct ColorTable
+    GDT_UInt32::ImageCore.N0f32,
 )
 
-eval(
-    @convert(
-        GDALDataType::DataType,
-        GDT_Unknown::Any,
-        GDT_Byte::UInt8,
-        GDT_UInt16::UInt16,
-        GDT_Int16::Int16,
-        GDT_UInt32::UInt32,
-        GDT_Int32::Int32,
-        GDT_Float32::Float32,
-        GDT_Float64::Float64,
-        GDT_CFloat32::ComplexF32,
-        GDT_CFloat64::ComplexF64
-    )
+@convert(
+    GDALDataType::DataType,
+    GDT_Unknown::Any,
+    GDT_Byte::UInt8,
+    GDT_UInt16::UInt16,
+    GDT_Int16::Int16,
+    GDT_UInt32::UInt32,
+    GDT_Int32::Int32,
+    GDT_Float32::Float32,
+    GDT_Float64::Float64,
+    GDT_CFloat32::ComplexF32,
+    GDT_CFloat64::ComplexF64
 )
 
-eval(
-    @convert(
-        OGRFieldType::GDAL.OGRFieldType,
-        OFTInteger::GDAL.OFTInteger,
-        OFTIntegerList::GDAL.OFTIntegerList,
-        OFTReal::GDAL.OFTReal,
-        OFTRealList::GDAL.OFTRealList,
-        OFTString::GDAL.OFTString,
-        OFTStringList::GDAL.OFTStringList,
-        OFTWideString::GDAL.OFTWideString,
-        OFTWideStringList::GDAL.OFTWideStringList,
-        OFTBinary::GDAL.OFTBinary,
-        OFTDate::GDAL.OFTDate,
-        OFTTime::GDAL.OFTTime,
-        OFTDateTime::GDAL.OFTDateTime,
-        OFTInteger64::GDAL.OFTInteger64,
-        OFTInteger64List::GDAL.OFTInteger64List,
-    )
+@convert(
+    OGRFieldType::GDAL.OGRFieldType,
+    OFTInteger::GDAL.OFTInteger,
+    OFTIntegerList::GDAL.OFTIntegerList,
+    OFTReal::GDAL.OFTReal,
+    OFTRealList::GDAL.OFTRealList,
+    OFTString::GDAL.OFTString,
+    OFTStringList::GDAL.OFTStringList,
+    OFTWideString::GDAL.OFTWideString,
+    OFTWideStringList::GDAL.OFTWideStringList,
+    OFTBinary::GDAL.OFTBinary,
+    OFTDate::GDAL.OFTDate,
+    OFTTime::GDAL.OFTTime,
+    OFTDateTime::GDAL.OFTDateTime,
+    OFTInteger64::GDAL.OFTInteger64,
+    OFTInteger64List::GDAL.OFTInteger64List,
 )
 
-eval(
-    @convert(
-        OGRFieldType::DataType,
-        OFTInteger::Bool,
-        OFTInteger::Int16,
-        OFTInteger::Int32,  # default type comes last
-        OFTIntegerList::Vector{Int32},
-        OFTReal::Float32,
-        OFTReal::Float64,  # default type comes last
-        OFTRealList::Vector{Float64},
-        OFTString::String,
-        OFTStringList::Vector{String},
-        OFTWideString::Nothing,
-        OFTWideStringList::Nothing,
-        OFTBinary::Vector{UInt8},
-        OFTDate::Dates.Date,
-        OFTTime::Dates.Time,
-        OFTDateTime::Dates.DateTime,
-        OFTInteger64::Int64,
-        OFTInteger64List::Vector{Int64},
-    )
+@convert(
+    OGRFieldType::DataType,
+    OFTInteger::Bool,
+    OFTInteger::Int16,
+    OFTInteger::Int32,  # default type comes last
+    OFTIntegerList::Vector{Int32},
+    OFTReal::Float32,
+    OFTReal::Float64,  # default type comes last
+    OFTRealList::Vector{Float64},
+    OFTString::String,
+    OFTStringList::Vector{String},
+    OFTWideString::Nothing,
+    OFTWideStringList::Nothing,
```

Comment:
(Just a comment that's unrelated to this PR)

This caught my eye (#231) as I was looking for examples where "default type comes last" is happening for the reverse mapping, and it looks like its the case in the current implementation too:

```julia
julia> convert(ArchGDAL.OGRFieldType, Nothing)
OFTWideStringList::OGRFieldType = 7
```